src/misc1.c

```diff
@@ -519,17 +519,20 @@ plines_m_win(win_T *wp, linenr_T first, linenr_T last, int limit_winheight)
 #endif
 	{
 #ifdef FEAT_DIFF
-	    if (first == wp->w_buffer->b_ml.ml_line_count)
-		count += diff_check_fill(wp, first + 1);
 	    if (first == wp->w_topline)
-		count += plines_win_nofill(wp, first, limit_winheight)
-							       + wp->w_topfill;
+		count += plines_win_nofill(wp, first, FALSE) + wp->w_topfill;
```

Comment:
I don't think it makes sense to limit each plines call for winheight. Instead exit the loop early when "w_height" is reached and return it when "limit_winheight" is TRUE.

Reply:
yes I think this makes sense. Thanks